kDriveCore/Utils/PHAsset/PHAssetIdentifier.swift

```diff
@@ -53,75 +78,110 @@ struct PHAssetIdentifier: PHAssetIdentifiable {
     }
 
     var baseImageSHA256: String? {
-        guard #available(iOS 15, *) else {
-            return nil
-        }
+        get throws {
+            guard #available(iOS 15, *) else {
+                return nil
+            }
 
-        let group = TolerantDispatchGroup()
-        var hash: String?
+            // We build an ExpiringActivity to track system termination
+            let uid = "\(#function)-\(asset.localIdentifier)-\(UUID().uuidString)"
+            let group = TolerantDispatchGroup()
+            var error: NSError?
+            var errorPointer = NSErrorPointer(&error)
```

Comment:
We can have a call to discuss this if you want @PhilippeWeidmann

Reply:
It took me some time to understand what you are trying to achieve but in the end it seems reasonable.
I think we can do better than NSErrorPointer though

Reply:
I would love to do without this.
This code is relying on a blocking `TolerantDispatchGroup` I do not see how a simple error in the delagate would work around the current synchronous locked code. We should DM.